cloudinit/config/cc_chef.py

```diff
@@ -6,78 +6,286 @@
+                },
+                'firstboot_path': {
+                    'type': 'string',
+                    'default': '/etc/chef/firstboot.json',
+                    'description': dedent("""\
+                        Path to write run_list and initial_attributes keys that
+                        should also be present in this configuration, defaults
+                        to ``/etc/chef/firstboot.json``.""")
+                },
+                'exec': {
+                    'type': 'boolean',
+                    'default':  False,
+                    'description': dedent("""\
+                        define if we should run or not run chef (defaults to
+                        false, unless a gem installed is requested where this
+                        will then default to true).""")
+                },
+                'client_key': {
+                    'type': 'string',
+                    'default': '/etc/chef/client.pem',
+                    'description': dedent("""\
+                        Optional path for client_cert. default to
+                        ``/etc/chef/client.pem``.""")
+                },
+                'encrypted_data_bag_secret': {
+                    'type': 'string',
+                    'default': None,
+                    'description': dedent("""\
+                        Specifies the location of the secret key used by chef
+                        to encrypt data items. By default, this path is set
+                        to None, meaning that chef will have to look at the
+                        path ``/etc/chef/encrypted_data_bag_secret`` for it.
+                        """)
+                },
+                'environment': {
+                    'type': 'string',
+                    'default': '_default',
+                    'description': dedent("""\
+                        Specifies which environment chef will use. By default,
+                        it will use the ``_default`` configuration."""),
+                },
+                'file_backup_path': {
+                    'type': 'string',
+                    'default': '/var/backups/chef',
+                    'description': dedent("""\
+                        Specifies the location in which backup files are
+                        stored. By default, it uses the
+                        ``/var/backups/chef`` location.""")
+                },
+                'file_cache_path': {
+                    'type': 'string',
+                    'default': '/var/cache/chef',
+                    'description': dedent("""\
+                        Specifies the location in which chef cache files will
+                        be saved. By default, it uses the ``/var/cache/chef``
+                        location.""")
+                },
+                'json_attribs': {
+                    'type': 'string',
+                    'default': '/etc/chef/firstboot.json',
```

Comment:
For any 'default values' that are defined elsewhere in a global, let's source those for the schema documentation so if they change in the future, docs are auto-updated.

```suggestion
                     'default': CHEF_RB_TPL_DEFAULTS['json_attribs'],
```

                    
```